app/serializers/basic_post_serializer.rb

```diff
@@ -44,8 +44,12 @@ def cooked
   end
 
   def ignored
-    object.is_first_post? && IgnoredUser.where(user_id: scope.current_user&.id,
-                                               ignored_user_id: object.user_id).exists?
+    object.is_first_post? &&
+      scope.current_user&.id != object.user_id &&
+      !object.user.admin? &&
+      !object.user.moderator? &&
+      IgnoredUser.where(user_id: scope.current_user&.id,
```

Comment:
I think I saw some discussion about this, but I'd like to confirm what we decided - isn't this a n+1 query here? Each serialized post will perform a query?

Reply:
So I think this is making one extra call for every post, and we can work on fixing/optimising this. But I don't think  that this is making N calls for every post we serialise. In my head, this feels more like: `1+1` rather than `N+1`.

If this is a `N+1` situation, it would be that for every _1_ **Post**, we are trying to fetch _N_ **Ignored Users**, which is commonly referred to as: `N+1` but it is actually a `1+N`.

Reply:
1+1 is a valid N+1 with N=1 😉

Reply:
Yup, it's fine. As long as you don't forget about it.

Reply:
If it is indeed making one query per post we have to fix this ASAP as that's a huge performance penalty on every topic.

(BTW this is exactly what is known as N+1 in rails. You are retrieving N posts to render in a topic. This code will cause N queries (1 per post) + 1 which is the query that retrieved the list of posts in the first place.)

Reply:
Ah totally my bad then. If it's only the first post it is indeed NOT a n+1 and I was wrong. But I think both @tgxworld and myself got confused by this. Even if we're both wrong in the end, I think a comment would save future programmers some time: `NOTE: This is not a n+1 because it's only on the first post`. Make it stupid proof :)

Reply:
I definitely think we should fix it, because if/when we remove that test, it'll become a bad N+1.

Reply:
In search, this can become an N+1 in the worst case scenario where every post is a first post.